### PR TITLE
Openc func

### PR DESCRIPTION
**Pull request by valichek**
Extend DialogService with openc func that returns promise for DialogController instance, the promise for result is stored in dialogController.result property

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@PWKad @jedd-ahyoung What do you guys think of this api?

File: src/dialog-service.js [hunk 1]
```diff
@@ -82,6 +53,68 @@ export class DialogService {
       return result;
     });
   }
+  
+  openc(settings?: Object): Promise<DialogController> {
```
Comment (PWKad) [inline src/dialog-service.js]: Can we give this a more descriptive name?

Comment (EisenbergEffect) [inline src/dialog-service.js]: Probably something like `openController()` yes?

**Comment by PWKad** [comment]
I like it could use a more descriptive name but otherwise I think this solves many questions that have been raised.  @valichek Can you add some tests to cover the new functionality?  Some pseudo examples - 

> expect openc to return the correct instance of the controller
> expect calling close on returned controller to resolve the promise
> expect controller.cancel to throw an error if the dialog is already closed

File: src/dialog-service.js (hunk 1 above)
Comment (PWKad) [inline src/dialog-service.js]: Not sure I was thinking that but it doesn't seem to properly describe what is being done, perhaps `openSynchronously`

Comment (EisenbergEffect) [inline src/dialog-service.js]: Yeah, I'm not sure. It's kind of tricky to name it.

**Comment by valichek** [comment]
@PWKad yes, I will cover it with tests, the only problem I have, is #190 not fixed. Actually this issue have raised when I started the PR

**Comment by PWKad** [comment]
@valichek understood I am guessing that is somehow related to this but reading that issue I'm not familiar with the changes that describes.  For my info, how is that blocking writing tests for this PR?

**Comment by valichek** [comment]
@PWKad sorry, just translation issue) it's blocking for the PR

**Comment by jedd-ahyoung** [comment]
This is a good addition, aside from the naming. I'd argue that this new API is actually better than the old one, as it's more flexible and can cover all of the use cases of the original "synchronous" API. Aside from naming and tests I think this should be good to go.

**Comment by EisenbergEffect** [comment]
@jedd-ahyoung Is this ok to merge? We can rename the method after merge.

**Comment by EisenbergEffect** [comment]
@jedd-ahyoung The release is tomorrow morning. Yes or no please.

**Pull request merged**